Makefile

```
@@ -0,0 +1,67 @@
+# This supports environments where $HOME/.cargo/env has not been sourced (CI, CLion Makefile runner)
+CARGO  = $(or $(shell which cargo),  $(HOME)/.cargo/bin/cargo)
+RUSTUP = $(or $(shell which rustup), $(HOME)/.cargo/bin/rustup)
+
+RUST_TOOLCHAIN := $(shell cat rust-toolchain)
+
+CARGO_OPTS := --locked
+CARGO := $(CARGO) $(CARGO_TOOLCHAIN) $(CARGO_OPTS)
+
+
+DISABLE_LOGGING = RUST_LOG=MatchesNothing
+
+.PHONY: all
+all: build
+
+.PHONY: build
+build:
+	$(CARGO) build $(CARGO_FLAGS)
+
+.PHONY: test
+test:
+	$(DISABLE_LOGGING) $(CARGO) test $(CARGO_FLAGS)
+
+.PHONY: check-format
+check-format:
+	$(CARGO) fmt --all -- --check
+
+.PHONY: format
+format:
+	$(CARGO) fmt --all
+
+.PHONY: lint
+lint:
+	$(CARGO) clippy --all-targets --all-features --workspace -- -D warnings -A renamed_and_removed_lints
+
+.PHONY: audit
+audit:
+	$(CARGO) audit
+
+.PHONY: build-docs-stable
+build-docs-stable: $(CRATES_WITH_DOCS_RS_MANIFEST_TABLE)
+
+doc-stable/%: CARGO_TOOLCHAIN += +stable
+doc-stable/%:
+	$(CARGO) doc $(CARGO_FLAGS) --manifest-path "$*/Cargo.toml" --no-deps
```

The target `build-docs-stable` doesn't build anything since we're missing the definition of `CRATES_WITH_DOCS_RS_MANIFEST_TABLE`.

I don't think docs are worth building or publishing here since they don't help a user make sense of how to use the binary.  That could possibly be fixed so we effectively just have a single doc page with similar content to the README, but I hadn't anticipated publishing the crate.

If we do need to test building the docs, I'd suggest we only need the following:
```suggestion
.PHONY: docs
docs:
	$(CARGO) doc $(CARGO_FLAGS) --no-deps
```